gammapy/catalog/tests/test_xmatch.py

```diff
@@ -10,13 +10,19 @@
     table_xmatch_circle_criterion,
     table_xmatch,
 )
+from ...utils.random import check_random_state
 
 
 def test_catalog_xmatch_circle():
+    # initialise random number generator
     np.random.seed(0)
+    #rng = check_random_state(0)
+
     catalog = make_catalog_random_positions_sphere(size=100, center='Milky Way')
     catalog['Source_Name'] = ['source_{:04d}'.format(_) for _ in range(len(catalog))]
     catalog['Association_Radius'] = Angle(10 * np.random.random(len(catalog)), unit='deg')
+    #catalog['Association_Radius'] = Angle(rng.uniform(0, 10, len(catalog)), unit='deg')
+    # TODO: not working with rng!!! (but why did it work before??!!!)
```

Comment:
If I use `check_random_state` at this point, the result is not reproducible and the test at the end fails.

But the more strange thing is: why was it reproducible before? The test uses `random` functions: `make_catalog_random_positions_sphere`, which makes use of `sample_sphere` and `sample_sphere_distance` without passing the `seed` as a parameter.
Was the `seed` passed implicitly, when setting it at the beginning of the test?

Ideas? I guess I need to pass the `random_state` parameter to `sample_sphere` and `sample_sphere_distance` all the way through `make_catalog_random_positions_sphere`, right?